.github/workflows/multi-arch-build.yaml

```diff
@@ -0,0 +1,119 @@
+name: build multi-arch images
+
+on:
+  push:
+    branches: [ master ]
+  schedule:
+  - cron:  '0 8 * * *'
+  # allows to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  multi:
+    name: multi-arch Buildah build
+    env:
+      BUILDAH_QUAY_REGISTRY: quay.io/buildah
+      CONTAINERS_QUAY_REGISTRY: quay.io/containers
+      # list of architectures for build
+      PLATFORMS: linux/amd64,linux/s390x,linux/ppc64le,linux/arm64
+
+    # build several images in parallel(upstream and stable)
+    strategy:
+      matrix:
+        # where the Dockerfiles for builds are located at contrib/buildahimage directory
+        source:
+        - upstream
+        - stable
+    runs-on: ubuntu-latest
+    # internal registry for temporary build before push
+    services:
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+          install: true
+
+      - name: Build and push local Buildah
+        uses: docker/build-push-action@v2
+        with:
+          context: contrib/buildahimage/${{ matrix.source }}
+          file: ./contrib/buildahimage/${{ matrix.source }}/Dockerfile
+          platforms: ${{ env.PLATFORMS }}
+          push: true
+          tags: |
+            localhost:5000/buildah/${{ matrix.source }}
+
+      # Simple verification that container works + get version for public image tag
+      - name: amd64 sniff test
+        id: run
+        env:
+          MATRIX_VERSION: ${{ matrix.source }}
+        run: |
+          # image tag is master for upstream source, actual version of buildah binary in the container for stable source
+          VERSION=v"$(docker run localhost:5000/buildah/${MATRIX_VERSION} buildah version | head -n 1 | awk '{print $2}')"
+          if [ "${MATRIX_VERSION}" == "upstream" ]; then
+            VERSION=master
+          fi
```

Comment:
This could be made slightly more reliable (less complex by default) by avoiding always running the docker command in all cases.  Consider this instead:

```bash
          VERSION=master
          if [ "${MATRIX_VERSION}" == "stable" ]; then
              VERSION=v"$(docker run localhost:5000/buildah/${MATRIX_VERSION} buildah version | head -n 1 | awk '{print $2}')"
          fi
```

Reply:
yeah, but then no test that container is functional for upstream case

Reply:
It's probably worth separating those two concerns as distinct steps.  One `buildah sniff test`, another `buildah version capture`. It will make it easier to debug as well, since output isn't potentially obscured/lost in the bash pipeline and stdout capture.  Maybe capture the complete stdout of `buildah version` in a test step, then (as a second step) parse into a new output, e.x. `echo ::set-output name=version::$(echo "${{ step.sniff.outputs.buildah }}" | head -n 1 | awk '{print $2}')`?

Reply:
there are 2 steps now, one to run the container and get buildah version, another one to extract image tag from it (+ generate labels for 2 push steps)